internal/client/job.go

```diff
@@ -325,6 +328,40 @@ func (c *Project) queueAndStreamJob(
 				stateEventTimer = nil
 			}
 
+			// Check if this job has been assigned a runner for the first time
+			if event.State != nil &&
+				event.State.Job != nil &&
+				event.State.Job.AssignedRunner != nil &&
+				assignedRunner == nil {
+
+				assignedRunner = event.State.Job.AssignedRunner
+
+				runner, err := c.client.GetRunner(ctx, &pb.GetRunnerRequest{RunnerId: assignedRunner.Id})
+				if err != nil {
+					c.UI.Output("Failed to inspect the runner (id %q) assigned for this operation: %s", assignedRunner.Id, err, terminal.WithErrorStyle())
+				}
+				switch runnerType := runner.Kind.(type) {
```

Comment:
I love this because it's like self-documenting of the different runners in our runner system.